File: lachancuoicgoi_flutter/lib/analysis/l3/core/gemini_response.dart
class AnalysisResponse {
  const AnalysisResponse({
    this.level,
    this.label,
    this.reason,
    this.recommendation,
  });

  final String? level;
  final String? label;
  final String? reason;
  final String? recommendation;

  factory AnalysisResponse.fromJson(Map<String, Object?> json) {
    return AnalysisResponse(
      level: json['level'] as String?,
      label: json['label'] as String?,
      reason: json['reason'] as String?,
      recommendation: json['recommendation'] as String?,
    );
  }
}

File: lachancuocgoi/app/src/main/java/com/example/lachancuocgoi/Analysis/L3/core/GeminiResponse.kt
package com.example.lachancuocgoi.Analysis.L3.core

import com.google.gson.annotations.SerializedName

/**
 * Response model cho analysis request.
 * Khớp với JSON format được định nghĩa trong prompt.
 */
data class AnalysisResponse(
    @SerializedName("level")
    val level: String?, // "green", "yellow", "orange", "red"

    @SerializedName("label")
    val label: String?, // Loại lừa đảo

    @SerializedName("reason")
    val reason: String?, // Giải thích ngắn gọn

    @SerializedName("recommendation")
    val recommendation: String? // Khuyến cáo ngắn gọn
)
